From 4b258c5a9078f8df60684ab7536ce3a8ff207e08 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 12 Oct 2017 10:03:57 +0000
Subject: [PATCH] configure.ac: fix configure failed while build dir contains "yes"

While the name of build dir contains "yes", the AC_EGREP_CPP
test always return true.

We rarely use "yes;" to name build dir, so s/yes/yes;/g
could fix the issue

Upstream-Status: Pending

Signed-off-by: Jian Kang <jian.kang@windriver.com>
---
 configure.ac | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index 12996cd..4765af6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -128,7 +128,7 @@ fi
 
 
 AC_CHECKING(for Ultrix)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#if defined(ultrix) || defined(__ultrix)
    yes;
 #endif
@@ -145,7 +145,7 @@ dnl ghazi@caip.rutgers.edu (Kaveh R. Ghazi):
 dnl BBN butterfly is not POSIX, but a MACH BSD system.
 dnl Do not define POSIX and TERMIO.
 AC_CHECKING(for butterfly)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#if defined(butterfly)
   yes;
 #endif
@@ -156,7 +156,7 @@ if test -n "$ULTRIX"; then
   test -z "$GCC" && CC="$CC -YBSD"
 fi
 AC_CHECKING(for POSIX.1)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#include <sys/types.h>
 #include <unistd.h>
 main () {
@@ -173,14 +173,14 @@ AC_TRY_COMPILE(
 #include <fcntl.h>], [int x = SIGCHLD | FNDELAY;], , AC_DEFINE(SYSV))
 
 AC_CHECKING(for sequent/ptx)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#ifdef _SEQUENT_
   yes;
 #endif
 ], LIBS="$LIBS -lsocket -linet";seqptx=1)
 
 AC_CHECKING(SVR4)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [main () {
 #if defined(SVR4) || defined(__SVR4)
   yes;
@@ -200,9 +200,9 @@ fi
 AC_CHECK_HEADERS([stropts.h string.h strings.h])
 
 AC_CHECKING(for Solaris 2.x)
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#if defined(SVR4) && defined(sun)
-  yes
+  yes;
 #endif
 ], LIBS="$LIBS -lsocket -lnsl -lkstat")
 
@@ -697,7 +697,7 @@ else
 pdir='/dev'
 fi
 dnl SCO uses ptyp%d
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#ifdef M_UNIX
    yes;
 #endif
@@ -880,7 +880,7 @@ fi
 )
 
 if test -z "$load" ; then
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#if defined(NeXT) || defined(apollo) || defined(linux)
   yes;
 #endif
@@ -1112,7 +1112,7 @@ AC_CHECKING(syslog in libbsd.a)
 AC_TRY_LINK(, [closelog();], AC_NOTE(- found.), [LIBS="$oldlibs"
 AC_NOTE(- bad news: syslog missing.) AC_DEFINE(NOSYSLOG)])])
 
-AC_EGREP_CPP(yes,
+AC_EGREP_CPP(yes;,
 [#ifdef M_UNIX
    yes;
 #endif
-- 
2.13.3

